starter-code/src/memory.js

```diff
@@ -1,15 +1,56 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function (cards) {
+  this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
+  this.currentPair = [];
+};
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
 
-// };
 
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+MemoryGame.prototype.shuffleCard = function (cardsArr) {
+    var m = cardsArr.length, t, i;
+    
+  // While there remain elements to shuffle…
+  while (m) {
 
-// }
+    // Pick a remaining element…
+    i = Math.floor(Math.random() * m--);
+    
+    // And swap it with the current element.
+    t = cardsArr[m];
+    cardsArr[m] = cardsArr[i];
+    cardsArr[i] = t;
+    //console.log(m,i,t);
+  }
 
-// MemoryGame.prototype.finished = function () {
+  return cardsArr;
 
-// };
+};
+
+MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+    var paired = false;
+    this.pairsClicked += 1;
+    $('#pairs_clicked').text(this.pairsClicked);
+    if(firstCard === secondCard){
+        $('.back').addClass('blocked');
+        this.pairsGuessed += 1;
+        this.currentPair = [];
+        paired = true;
+        $('#pairs_guessed').text(this.pairsGuessed);
+    }else{
+        $('.back').addClass('blocked');
```

Comment:
The only thing you are missing is setTimeout() to delay changing the background back to blue.
```
else{
        $('.back').addClass('blocked');
        setTimeout(function(){
            $('.just-clicked').css("background", "#456783")
            $('.just-clicked').removeClass('just-clicked');
            $('.back').removeClass('blocked');
        }, 1000)
    }
```